nixos/modules/services/cluster/k3s/default.nix

```
@@ -113,8 +110,8 @@ in
         ExecStart = concatStringsSep " \\\n " (
           [
             "${cfg.package}/bin/k3s ${cfg.role}"
-          ] ++ (optional cfg.docker "--docker")
-          ++ (optional (cfg.docker && config.systemd.enableUnifiedCgroupHierarchy) "--kubelet-arg=cgroup-driver=systemd")
+          ]
+          ++ (optional (config.systemd.enableUnifiedCgroupHierarchy) "--kubelet-arg=cgroup-driver=systemd")
```

```suggestion
          ++ (optional (config.systemd.enableUnifiedCgroupHierarchy) "--kubelet-arg=cgroup-driver=systemd")
```

@euank Should this be removed?

---

I think that this should be maintained.

Using systemd as cgroup driver keeps a single view of resources in a node, otherwise you end up with two, systemd for system process and cgroupfs for K8S.

Note that NixOS enables this by default, so we should rely on systemd only by default.

- https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/configure-cgroup-driver/
- https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/system/boot/systemd.nix#L603